components/script/dom/webglrenderingcontext.rs

```diff
@@ -1156,6 +1111,111 @@ impl WebGLRenderingContext {
             }
         }
     }
+
```

Comment:
It would be nice if the refactors were split out from the functional content of the commit.